src/main/kotlin/anttracker/Contact.kt

```diff
@@ -148,26 +100,79 @@ fun enterContactInformation(): Contact? {
         if (department.isBlank()) {
             department = ""
         }
+        break
+    }
 
-        val contact =
-            transaction {
-                Contact.new {
-                    this.name = name
-                    this.phoneNumber = phone
-                    this.email = email
-                    this.department = department
-                }
-            }
+    return transaction {
+        Contact.new {
+            this.name = name
+            this.phoneNumber = phone
+            this.email = email
+            this.department = department
+        }
+    }
+}
 
-        println("Contact ${contact.name} has been created.")
-        return contact
+// Checks if the contact name is unique in the database
+private fun isUniqueContactName(name: String): Boolean {
+    return transaction {
+        Contact.find { Contacts.name eq name }.empty()
     }
 }
 
 // ----------------------------------------------------------------------------
-// Returns all information about the contact identified by a given name.
+// A class that represents a page of contacts.
+// This class handles pagination and display of contacts from the database.
 // ---
-fun getContactInfo(name: String): Contact? =
-    transaction {
-        Contact.find { Contacts.name eq name }.firstOrNull()
+private class PageOfContact : PageOf<Contact>(Contact) {
+    init {
+        initLastPageNum()
+    }
+
+    override fun getQuery(): Query =
+        Contacts
+            .selectAll()
+            .orderBy(
+                Contacts.name to SortOrder.ASC,
+            )
+
+    override fun printRecord(record: Contact) {
+        println(record.name)
+    }
+}
+
+// Display pages of contacts to console and select one
+fun selectContact(): Contact? {
+    val contactPage = PageOfContact()
+    contactPage.loadRecords() // Adjust limit and offset as necessary
+    contactPage.display()
+
+    var linenum: Int? = null
+    while (linenum == null) {
+        println("Please select contact. ` to abort:")
+        val userInput = readln()
+        when (userInput) {
+            "`" -> return null
+            "" -> {
+                if (!contactPage.lastPage()) {
+                    contactPage.loadNextPage()
+                    contactPage.display()
+                }
+            }
+            else -> {
+                try {
+                    // Check if the line number is valid and within the current page's range
+                    val userInputInt = userInput.toInt()
+                    if (userInputInt in 1..20 && userInputInt <= contactPage.recordsSize()) {
```

Comment:
Instead of checking against this range, you can use the new `contactPage.isValidLineNum` function! Like so:

```kotlin
val userInputInt = selection.toInt()

// validate integer selection
if (issuePage.isValidLineNum(userInputInt)) {
    linenum = userInputInt
}
```